From e4c3bd27f14834579998f94868651dc7da07a654 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:31 -0400
Subject: [usb] ohci: ohci_init_driver() - sanity check overrides

Message-id: <1410887463-20674-20-git-send-email-dzickus@redhat.com>
Patchwork-id: 95097
O-Subject: [RHEL7 PATCH 019/591] USB: OHCI: ohci_init_driver(): sanity check overrides
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit c80ad6d1cd6c6662d0cff752d94a1a9fde6de4ac
Author: Kevin Hilman <khilman@linaro.org>
Date:   Fri Sep 27 08:10:32 2013 -0700

    USB: OHCI: ohci_init_driver(): sanity check overrides

    Check for non-NULL overrides before dereferencing since platforms may
    pass in NULL overrides.

    Signed-off-by: Kevin Hilman <khilman@linaro.org>
    Acked-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Tomasz Figa <tomasz.figa@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ohci-hcd.c b/drivers/usb/host/ohci-hcd.c
index 604cad1..db48b27 100644
--- a/drivers/usb/host/ohci-hcd.c
+++ b/drivers/usb/host/ohci-hcd.c
@@ -1161,10 +1161,12 @@ void ohci_init_driver(struct hc_driver *drv,
  /* Copy the generic table to drv and then apply the overrides */
  *drv = ohci_hc_driver;
 
- drv->product_desc = over->product_desc;
- drv->hcd_priv_size += over->extra_priv_size;
- if (over->reset)
-  drv->reset = over->reset;
+ if (over) {
+  drv->product_desc = over->product_desc;
+  drv->hcd_priv_size += over->extra_priv_size;
+  if (over->reset)
+   drv->reset = over->reset;
+ }
 }
 EXPORT_SYMBOL_GPL(ohci_init_driver);
 
-- 
1.7.1